apps/explorer/src/api/assets/assetsApi.ts

```diff
@@ -164,10 +164,23 @@ export const assetsQueries = {
     item: (dataUri: string, assetId: string) =>
       queryOptions({
         queryKey: ['nftData', dataUri],
-        queryFn: async (): Promise<NFTTokenUriMetaData & { assetId: string }> => {
+        // TODO: Should the image field in NFTTokenUriMetaData be optional?
+        queryFn: async (): Promise<Optional<NFTTokenUriMetaData, 'image'> & { assetId: string }> => {
           const nftData = (await axios.get(dataUri)).data as NFTTokenUriMetaData
 
-          return matchesNFTTokenUriMetaDataSchema(nftData) ? { ...nftData, assetId } : Promise.reject()
+          if (!nftData || !nftData.name) {
+            return Promise.reject()
+          }
+
+          return matchesNFTTokenUriMetaDataSchema(nftData)
+            ? { ...nftData, assetId }
+            : nftData.name
+              ? {
+                  assetId,
+                  name: nftData.name,
+                  image: nftData.image ? nftData.image.toString() : undefined
```

Comment:
here you need to first check if `image` exists.

```suggestion
                  image: nftData?.image ? nftData.image.toString() : undefined
```

Reply:
You mean `nftData`? Mhh at that point we know it exists because of the check line 171 above, no?

Reply:
You are right. We just don't know if `image` exists. My bad